backend/de.metas.banking/de.metas.banking.camt53/src/main/java/de/metas/banking/camt53/process/C_BankStatement_Camt53_ImportAttachment.java

```diff
@@ -0,0 +1,165 @@
+
+	private final BankStatementCamt53Service bankStatementCamt53Service = SpringContextHolder.instance.getBean(BankStatementCamt53Service.class);
+	private final AttachmentEntryService attachmentEntryService = SpringContextHolder.instance.getBean(AttachmentEntryService.class);
+	private final BankStatementImportFileService bankStatementImportFileService = SpringContextHolder.instance.getBean(BankStatementImportFileService.class);
+
+	@Param(parameterName = I_AD_AttachmentEntry.COLUMNNAME_AD_AttachmentEntry_ID, mandatory = true)
+	private AttachmentEntryId p_AD_AttachmentEntry_ID;
+
+	@Override
+	public ProcessPreconditionsResolution checkPreconditionsApplicable(@NonNull final IProcessPreconditionsContext context)
+	{
+		if (context.isNoSelection())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNoSelection();
+		}
+
+		if (!context.isSingleSelection())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNotSingleSelection();
+		}
+
+		if (isSelectedRecordProcessed(context.getSingleSelectedRecordId()))
+		{
+			return ProcessPreconditionsResolution.reject(msgBL.getTranslatableMsgText(MSG_SELECTED_RECORD_ALREADY_PROCESSED));
+		}
+
+		return ProcessPreconditionsResolution.accept();
+	}
+
+	@Override
+	protected String doIt()
+	{
+		final AttachmentEntryDataResource data = attachmentEntryService.retrieveDataResource(getAttachmentEntryId());
+
+		final Set<BankStatementId> importedBankStatementIds = bankStatementCamt53Service.importBankToCustomerStatement(data.getInputStream());
+
+		openImportedRecords(importedBankStatementIds);
+
+		markRecordAsProcessed(data.getFilename());
+
+		return JavaProcess.MSG_OK;
+	}
+
+	private void markRecordAsProcessed(@NonNull final String filename)
+	{
+		final BankStatementImportFile bankStatementImportFile = buildBankStatementImportFile(filename);
+		
+		bankStatementImportFileService.save(bankStatementImportFile);
+	}
+
+	@NonNull
+	private BankStatementImportFile buildBankStatementImportFile(@NonNull final String filename)
+	{
+		return BankStatementImportFile.builder()
+				.bankStatementImportFileId(BankStatementImportFileId.ofRepoId(getRecord_ID()))
+				.filename(filename)
+				.importedTimestamp(Instant.now())
+				.processed(true)
+				.build();
+	}
+	
+	private void openImportedRecords(@NonNull final Set<BankStatementId> importedBankStatementIds)
+	{
+		if (importedBankStatementIds.size() == 0)
+		{
+			final ITranslatableString msg = msgBL.getTranslatableMsgText(MSG_NO_STATEMENT_IMPORTED);
+			throw new AdempiereException(msg)
+					.markAsUserValidationError();
+		}
+		else if (importedBankStatementIds.size() == 1)
+		{
+			setRecordToOpen(importedBankStatementIds.iterator().next());
+		}
+		else
+		{
+			setRecordsToOpen(importedBankStatementIds);
+		}
+	}
+
+	private void setRecordsToOpen(@NonNull final Set<BankStatementId> importedBankStatementIds)
```

Comment:
loud thinking: why not always open the result in a new tab?
In case there is only on bank statement we could use SingleDocument target.
In case there are more, we could go with GridView.